### ISSUE TITLE
Q: backport patches for v2.4.3

### ISSUE THREAD

**Issue by pcmoore** [seccomp/libseccomp#187]
It is looking like we will likely need to make a v2.4.3 release, so let's track the required backports in this issue.

**Comment by pcmoore** [comment]
Two PRs immediately come to mind:
* PR #186: fix a missing ppoll syscall define
* PR #182 
  * commit which fixes 15-basic-resolver.c
  * commit which adds the missing 51-live-user_notification.py test (optional)

**Comment by ncopa** [comment]
I think it would be great if we could also include the performance regression fixes from PR #180 (commits 21b98d85e8bfdb701a5f9afd54ff5175af910a45 and 19af04da86e9a4168a443f3563fc7aec8839edf0). Those patches applies cleanly to current v2.4.2 and we already backported them to alpine linux's edge (https://github.com/alpinelinux/aports/commit/9d67c32ea232c56ebe4ea9b62207222a94e898a3) and I want suggest to backport it to ubuntu as well.

This is a problem that prevents customers/users to upgrade to 2.4 and get the security update.

**Comment by pcmoore** [comment]
Hi @ncopa, I understand your concern with the performance of the v2.4.z release, but I worry that the changes in PR #180 are too significant for patch release.  The changes in PR #180 will be present in v2.5.0.

**Comment by drakenclimber** [comment]
> I worry that the changes in PR #180 are too significant for patch release.

I think that's the right answer as well.  But with RISC V support coming, the binary tree, and the performance improvements cited above, I think we have a strong case for putting more emphasis on getting the v2.5 release out the door.

**Comment by pcmoore** [comment]
@drakenclimber we really need to get a v2.4.3 release out too.  Some time ago I recall us talking about you leading the charge on the maintenance releases, want to wrap up the two remaining v2.4.3 issues?

**Comment by drakenclimber** [comment]
> @drakenclimber we really need to get a v2.4.3 release out too. Some time ago I recall us talking about you leading the charge on the maintenance releases, want to wrap up the two remaining v2.4.3 issues?

Doggonit... I need a time machine! :)  Yeah, I'll get working on v2.4.3 first thing next week.

**Comment by pcmoore** [comment]
> Doggonit... I need a time machine! :)

You and me both :)

**Comment by anthraxx** [comment]
Awesome news that 2.4.3 is progressing :smile_cat: 

Just a tiny reminder, could we get just one more release be signed by @pcmoore so we have a cryptographic proof of @drakenclimber key, which is now embedded in the repo itself but yet the commit chain is unsigned? That would be absolutely awesome :tada: 

PS: Congrats at @drakenclimber for the new duties :smile:

**Comment by pcmoore** [comment]
Hi @anthraxx, we discussed this quite a bit over email, @drakenclimber will be managing and signing the "stable" releases, e.g. v2.4.3, and I will be managing and signing the major/minor releases, e.g. v2.5.0.  We have updated the README.md with our key fingerprints and only @drakenclimber and myself have write access to this repo on GitHub.  If that assurance isn't sufficient you may need to wait until the v2.5.0 release.

**Comment by anthraxx** [comment]
Well, write access to github != cryptographic proof which is the very requirement for appropriate trust graph handling when dealing with signing keys, otherwise you don't even need any signing keys if your trust model is based on the pure access control to github.
If this was only planned for v2.5.0 then we clearly had a different understanding of the discussion. This currently becomes problematic in downstream consumers as there are essential fixes missing related to latest linux syscalls.

Anyway, If you could git sign the next v2.4.3 tag, like you git signed the v2.4.2, that would already be sufficient to have an embedded implicit trust of any content within and that would finally legitimate the new key.

**Comment by pcmoore** [comment]
Once again @anthraxx, we've discussed all of this at length over email, if you are not satisfied with the current situation you may need to wait until v2.5.0.

**Comment by anthraxx** [comment]
It's not be, its a shitload of users that are blocked from having this update, all that because a simple cryptographic proof is too much hassle.
You have signed the git tag of v2.4.2 yourself, so I don't see what the problem is in kindly asking to also git sign the future v2.4.3, I'm not talking about the release tarball.

**Comment by pcmoore** [comment]
> It's not me ...

Since the v2.4.2 release, over three months ago, you are the only person and/or downstream packager who has come to us with this concern.  If you want to block Arch, and its users, from receiving the update, that is clearly your choice to make.  I have informed you of our policy multiple times now, and while you may not accept my current stance on this issue, I can assure you that your continued attempts to sway me have reached the point where I'm finding them inflammatory and not at all persuasive.

While I agree that the GH README.md update is not as cryptographically strong as a signed git tag and/or release tarball, the tags/tarballs signed by me have no special guarantees of provenance other than my word/declaration.  Even that declaration is only considered special because I am the original author and the first one to start signing the tags/tarballs.  @drakenclimber has signed the v2.4.2 release, both the git tag and the tarball, same as I did the original signed libseccomp release (although I didn't start signing git tags until the v2.4.0 release).

You are welcome to comment further on this @anthraxx, but this comment is probably the last I will make on this matter.

**Comment by eli-schwartz** [comment]
@pcmoore, there is no cryptographically secure way of verifying that the following didn't happen:

- your github account was compromised by an unauthorized user 
- your email address was compromised by an unauthorized user
- the unauthorized user has been communicating with people, claiming that some new person is authorized to sign releases
- the Github Releases tarball for the signed git tag was deleted and replaced with something signed by the theoretical attacker
- an unauthenticated git commit was pushed to master, claiming that the new key is valid

It's unfortunate that you don't use `git config commit.gpgsign true` to provide a cryptographic chain of trust for git master. If at any point, a commit was signed by your PGP signing key, then trust would be established that the previous commits, including e5c4b5ee8086b936d28b080e5477950f632b4d43, are valid. As would using `gpg --sign-key` to attest your trust in the new signing key, or posting a PGP-signed message to the same effect.

The fact that it's so simple to do one of these things, and you haven't done so, is suspicious. More suspicious than the initial changeover itself.

**Comment by anthraxx** [comment]
1. No, you personally have signed v2.4.2
```
git tag --verify v2.4.2                                              
object 1b6cfd1fc0b7499a28c24299a93a80bd18619563
type commit
tag v2.4.2
tagger Paul Moore <paul@paul-moore.com> 1573163595 -0500

version 2.4.2
gpg: Signature made Thu 07 Nov 2019 10:53:15 PM CET
gpg:                using RSA key 4B42A8CF07F299D554977EEFEA20F2DA97378973
gpg:                issuer "paul@paul-moore.com"
gpg: Good signature from "Paul Moore <paul@paul-moore.com>" [unknown]
gpg:                 aka "Paul Moore <pcmoore@umich.edu>" [unknown]
gpg:                 aka "Paul Moore <pmoore@redhat.com>" [unknown]
Primary key fingerprint: 7100 AADF AE6E 6E94 0D2E  0AD6 55E4 5A5A E8CA 7C8A
     Subkey fingerprint: 4B42 A8CF 07F2 99D5 5497  7EEF EA20 F2DA 9737 8973

```

2. There aren't too much distros that have picked up the latest release so far. Debian does. but they also use the git repo in a mirror + add kind of concept and as shown above the repo tag is done by you.
Other distros like fedora, nix, void, and alpine that have the latest version are all not using any signature checking at all. It's not my fault that they don't :man_shrugging:

3. You can also simply sign his key and upload the signature to a keyserver if the drama is all about who does which release.

**Comment by drakenclimber** [comment]
Merged to the release-2.4 branch in commit https://github.com/seccomp/libseccomp/commit/c09346bb9847c2a1506bb5ed55afcf78d199bea6
